cmd/kubelet/app/server.go

```diff
@@ -440,6 +444,115 @@ func InitializeTLS(s *options.KubeletServer) (*server.TLSOptions, error) {
 	return tlsOptions, nil
 }
 
+// createCertAndKey tries to:
+// Use the boostrap auth token to get x509 certs if the token is set.
+// If this fails or the token is empty, then self-generates cert and key pair.
+func createCertAndKey(s *options.KubeletServer) error {
+	var err error
+	if s.BootstrapAuthToken != "" {
+		if err = getCertFromAPIServer(s); err == nil {
+			glog.V(4).Infof("Using cert got from the API server (%s, %s)", s.TLSCertFile, s.TLSPrivateKeyFile)
+			return nil
+		}
+		glog.Warningf("Cannot get cert from API server: %v, try to generate self-signed cert", err)
```

Comment:
Is this the correct failure mode?

If a bootstrap token is provided, but doesn't work, that indicates a configuration error on the operator's part so perhaps it's better to just error out rather than do extra work to generate a self-signed cert and potentially end up in a confusing state.

cc @gtank, what's the right behaviour there IYO?